Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Systems/ArtifactGasTriggerSystem.cs

```diff
@@ -0,0 +1,53 @@
+using Content.Server.Atmos.EntitySystems;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Triggers.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Random;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts.Triggers.Systems;
+
+public class ArtifactGasTriggerSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly AtmosphereSystem _atmosphereSystem = default!;
+    [Dependency] private readonly ArtifactSystem _artifactSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ArtifactGasTriggerComponent, ComponentInit>(OnInit);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        var query = EntityManager.EntityQuery<ArtifactGasTriggerComponent>();
+        foreach (var component in query)
+        {
+            if (component.ActivationGas == null)
+                return;
+
+            var transform = Transform(component.Owner);
```

Comment:
Just use the overload with multiple components i.e. EntityQuery<ArtifactGasTriggerComponent, TransformComponent>